### ISSUE TITLE
Invalid TypeError when calling `reverse ∘ toStringOrNull` on a node

### ISSUE THREAD

**Issue by CelineWuest** [neo4j/neo4j#13152]
When running the following query:
```cypher
CREATE (n) RETURN reverse(toStringOrNull(n))
```

I encountered the following error:
```
Neo4jError: Neo.ClientError.Statement.TypeError (Invalid input for function 'reverse()': Expected a string or a list; consider converting the value to a string with toString() or creating a list.)
```

I believe the query mentioned above is semantically and syntactically correct and thus no error should be thrown here.
Additionally, alternatives that evaluate to the same qyery run successfully, like for example:
`CREATE (n) RETURN reverse(null)` or
`CREATE (n) WITH toStringOrNull(n) AS x RETURN reverse(x)`

I encountered this issue when testing queries against the **Neo4j 5.6.0 enterprise version** in a Docker container running **alpine v.3**.
The query runs successfully in the community version and returns `null`.

### Steps to reproduce
Run the following query and observe that an error is thrown:
```cypher
CREATE (n) RETURN reverse(toStringOrNull(n))
```

### Expected behavior
The query should run successfully and return `null`.

### Actual behavior
Community Version:  The query runs successfully and returns `null`.
Enterprise Version: The query fails with the error message `Neo4jError: Neo.ClientError.Statement.TypeError (Invalid input for function 'reverse()': Expected a string or a list[..]`

**Comment by loveleif** [comment]
Thanks @DominicWuest! This issue is fixed in an open PR and is scheduled to be released in 5.8.

**Comment by loveleif** [comment]
PR merged and scheduled for release in 5.8.0. Please reopen or create a new issue if the problem should reappear despite our best efforts.
